File: java/socialbot-examples/Post.java
import java.sql.Timestamp;

public class Post {
    private int id;
    private String message = "";
    private User user;
    private User wall;
    private Timestamp publishingDate;
    private User[] likedBy = new User[0];

    public Post(int id, String message, User user, User wall, Timestamp publishingDate, User[] likedBy) {
        this.id = id;
        this.message = message;
        this.user = user;
        this.wall = wall;
        this.publishingDate = publishingDate;
        this.likedBy = likedBy;
    }

    /** @return the id */
    public int getId() {
        return id;
    }

    /** @return the message */
    public String getMessage() {
        return message;
    }

    /** @return the user */
    public User getUser() {
        return user;
    }

    /** @return the userid */
    public int getUserId() {
        return user.getId();
    }

    /** @return the username */
    public String getUsername() {
        return user.getUsername();
    }

    /** @return the wall */
    public User getWall() {
        return wall;
    }

    /** @return the publishingDate */
    public Timestamp getPublishingDate() {
        return publishingDate;
    }

    /** @return the likedBy */
    public User[] getLikedBy() {
        return likedBy;
    }
}

File: python/socialbot-example/Post.py
#!/usr/bin/python
from typing import List

from User import *


class Post:
    def __init__(self, id: int, message: str, user: User, wall: User, publishingDate: str, likedBy: List[User]):
        """ erzeugt ein Post-Objekt, das über Methoden die Informationen zurückgibt """
        self._id = id
        self._message = message
        self._user = user
        self._wall = wall
        self._publishingDate = publishingDate
        self._likedBy = likedBy

    @property
    def id(self) -> int:
        return self._id

    @property
    def message(self) -> str:
        return self._message

    @property
    def user(self) -> User:
        return self._user

    @property
    def userId(self) -> id:
        return self._user.id

    @property
    def username(self) -> str:
        return self._user.username

    @property
    def wall(self) -> User:
        return self._wall

    @property
    def publishingDate(self) -> str:
        return self._publishingDate

    @property
    def likedBy(self) -> List[User]:
        return self._likedBy
